crates/fuel-core/src/executor.rs

```diff
@@ -979,21 +962,47 @@ where
             ExecutionTypes::Production(tx) => {
                 for input in tx.inputs_mut() {
                     match input {
-                        Input::CoinSigned { tx_pointer, .. }
-                        | Input::CoinPredicate { tx_pointer, .. } => {
-                            let _ = tx_pointer;
-                            // TODO: Also calculate `tx_pointer` based on utxo's pointer.
-                            // *tx_pointer = TxPointer::new(height.into(), idx);
+                        Input::CoinSigned {
+                            tx_pointer,
+                            utxo_id,
+                            owner,
+                            amount,
+                            asset_id,
+                            maturity,
+                            ..
+                        }
+                        | Input::CoinPredicate {
+                            tx_pointer,
+                            utxo_id,
+                            owner,
+                            amount,
+                            asset_id,
+                            maturity,
+                            ..
+                        } => {
+                            let coin = self.get_coin_or_default(
+                                db,
+                                *utxo_id,
+                                *owner,
+                                *amount,
+                                *asset_id,
+                                (*maturity).into(),
+                            )?;
+                            *tx_pointer = TxPointer::new(
+                                coin.block_created.into(),
+                                coin.block_created_tx_idx,
+                            );
                         }
                         Input::Contract {
                             ref mut utxo_id,
                             ref mut balance_root,
                             ref mut state_root,
+                            ref mut tx_pointer,
                             ref contract_id,
                             ..
                         } => {
                             let mut contract = ContractRef::new(&mut *db, *contract_id);
-                            *utxo_id =
+                            (*utxo_id, *tx_pointer) =
                                 contract.validated_utxo(self.config.utxo_validation)?;
                             *balance_root = contract.balance_root()?;
                             *state_root = contract.state_root()?;
```

Comment:
Please feel free to remove the comment below this line(I can't leave a comment there).

Reply:
done